packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -460,18 +474,18 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
     }
 
     /**
-     * This tells whether the used state of this node has changed since last successful summary. If the used routes
-     * of this node changed, its used state is considered changed. Basically, if this node or any of its child nodes
-     * was previously used and became unused (or vice versa), its used state has changed.
+     * This tells whether the used state of this node has changed since last successful summary. Since the last GC run,
+     * if this node's used route changed or its unreferenced state was reset, its used state is considered changed.
      */
     private hasUsedStateChanged(): boolean {
         // If GC is disabled, we are not tracking used state, return false.
         if (this.gcDisabled) {
             return false;
         }
 
-        return this.referenceUsedRoutes === undefined ||
-            JSON.stringify(this.usedRoutes) !== JSON.stringify(this.referenceUsedRoutes);
+        return this.referenceUsedRoutes === undefined
+            || JSON.stringify(this.usedRoutes) !== JSON.stringify(this.referenceUsedRoutes)
+            || this.stateResetSinceLastRun;
```

Comment:
It's easier to follow code if terms used to describe things are the same.
It feels like hasUsedStateChanged() and stateResetSinceLastRun trying to answer same question, no?